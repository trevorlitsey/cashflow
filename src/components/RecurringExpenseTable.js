import React from 'react';
import { array } from 'prop-types'
import { format } from 'date-fns';
import { List, Avatar } from 'antd';

class RecurringExpenseTable extends React.Component {

	static propTypes = {
		recurringExpenses: array.isRequired,
	}

	render() {

		const { recurringExpenses } = this.props

		return (
			<List
				itemLayout="horizontal"
				dataSource={recurringExpenses || []}
				renderItem={item => (
					<List.Item actions={[<a>edit</a>, <a>delete</a>]}>
						<List.Item.Meta
							avatar={<Avatar icon="calendar" />}
							title={<a href="https://ant.design">{item.name}</a>}
							description={`every ${item.frequency} ${item.interval} starting on ${format(item.startDate, 'MMMM D, YYYY')}`}
						/>
					</List.Item>
				)}
			/>
		)
	}
}

export default RecurringExpenseTable;