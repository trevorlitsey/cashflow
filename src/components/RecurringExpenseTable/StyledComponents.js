import styled from 'styled-components';

export const Container = styled.div`
	width: 100%;
`

export const UnOrderedList = styled.ul`
	
	padding: 0;
	display: block;
	width: 100%;
	
	li {
		margin: 10px 0;
		padding: 0 10px 10px;
		display: grid;
		grid-template-columns: 4fr 1fr;
		border-bottom: 1px solid HSLA(220, 8%, 92%, 1.00);
	}

	p {
		margin: 5px;
	}

	a {
		margin: 0 4px;
	}

	.right {
		display: flex;
		align-items: center;
		justify-content: flex-end;

		& > span {
			cursor: pointer;
			color: HSLA(220,0%,70%,1.00);
		}
	}

	.title {
		font-weight: 600;
	}
`

export const NoneYet = styled.p`
	text-align: center;
	color: grey;
`