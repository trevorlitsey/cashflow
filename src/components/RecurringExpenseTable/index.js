import React from 'react';
import { object, func } from 'prop-types'
import { format } from 'date-fns';
import { List, Avatar, Popconfirm, message } from 'antd';
import currencyFormatter from 'currency-formatter';

import { convertObjToArr } from '../../helpers';

class RecurringExpenseTable extends React.Component {

	static propTypes = {
		recurringExpenses: object.isRequired,
		deleteRecurringExpense: func.isRequired,
	}

	handleRecurringExpenseDelete = (e, id) => {
		e.preventDefault();
		this.props.deleteRecurringExpense(id)
		message.success('recurring income/expense deleted');
		return true;
	}

	render() {

		const { recurringExpenses, deleteRecurringExpense } = this.props

		return (
			<List
				itemLayout="horizontal"
				dataSource={convertObjToArr(recurringExpenses) || []}
				renderItem={item => (
					<List.Item actions={[
						<a>edit</a>,
						<Popconfirm title="Are you sure delete this expense?" onConfirm={(e) => this.handleRecurringExpenseDelete(e, item.id)} okText="Yes" cancelText="No">
							<a>delete</a>
						</Popconfirm>
					]}>
						<List.Item.Meta
							avatar={<Avatar icon="calendar" />}
							title={`${item.name} (${currencyFormatter.format(item.amount, { code: 'USD', precision: 0 })})`}
							description={`every ${item.frequency} ${item.interval} starting on ${format(item.startDate, 'MMMM D, YYYY')}`}
						/>
					</List.Item>
				)}
			/>
		)
	}
}



export default RecurringExpenseTable;