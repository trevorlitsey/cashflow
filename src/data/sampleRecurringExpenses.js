const recurringExpenses = {
	one: {
		name: 'ya!',
		startDate: 1522698884359,
		name: 'an expenses',
		frequency: 2,
		interval: 'days',
	},
	two: {
		name: 'check this out',
		startDate: 1522698884359,
		name: 'an expenses',
		frequency: 1,
		interval: 'weeks',
	}
}

export default recurringExpenses;