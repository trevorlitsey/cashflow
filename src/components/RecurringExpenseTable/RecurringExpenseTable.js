import React from 'react';
import { object, func } from 'prop-types'
import { format } from 'date-fns';
import { Popconfirm, message, Tooltip } from 'antd';
import currencyFormatter from 'currency-formatter';

import { SubTitle } from '../../styles/SharedComponents';
import { Container, UnOrderedList, NoneYet } from './StyledComponents';

import { convertObjToArr } from '../../helpers';

// -------- TODO ---------
// - allow edit

class RecurringExpenseTable extends React.Component {

	static propTypes = {
		recurringExpenses: object.isRequired,
		deleteRecurringExpense: func.isRequired,
	}

	handleRecurringExpenseDelete = (id) => {
		this.props.deleteRecurringExpense(id)
		message.success('recurring income/expense deleted');
	}

	render() {

		const { recurringExpenses, deleteRecurringExpense } = this.props

		if (Object.keys(recurringExpenses).length === 0) {
			return (
				<div>
					<SubTitle>Recurring Income/Expenses:</SubTitle>
					<NoneYet>none yet</NoneYet>
				</div>
			)
		}

		return (
			<Container>
				<SubTitle>Recurring Income/Expenses:</SubTitle>
				<UnOrderedList>
					{convertObjToArr(recurringExpenses)
						.sort((a, b) => a.startDate - b.startDate) // sort asc
						.map((values) => {

							const { name, amount, startDate, frequency, interval, id } = values;

							return (
								<li key={id}>
									<div className="left">
										<p className="title">{`${name} (${currencyFormatter.format(amount, { code: 'USD', precision: 0 })})`}</p>
										<p className="subtitle">{`every ${frequency} ${interval} starting on ${format(startDate, 'MMMM D, YYYY')}`}</p>
									</div>
									<div className="right">
										<Tooltip data-test="delete" onClick={() => this.handleRecurringExpenseDelete(id)} placement="left" title="delete recurring income/expense"><span>x</span></Tooltip>
									</div>
								</li>
							)
						})}
				</UnOrderedList>
			</Container>
		)
	}
}

export default RecurringExpenseTable;