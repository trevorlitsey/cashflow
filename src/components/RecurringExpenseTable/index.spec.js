import React from 'react';
import { shallow } from 'enzyme';

import sampleRecurringExpenses from '../../data/sampleRecurringExpenses';

import RecurringExpenseTable from './index';

describe('RecurringExpenseTable', () => {

	it('should handleRecurringExpenseDelete without error', () => {
		const instance = renderRecurringExpenseTable().instance();

		const event = {
			preventDefault: () => { },
		}
		const id = 567876;
		instance.handleRecurringExpenseDelete(event, id);

		expect(instance.props.deleteRecurringExpense.mock.calls.length).toBe(1);
	})

})

function renderRecurringExpenseTable(props = {}) {
	const propsToUser = {
		recurringExpenses: sampleRecurringExpenses,
		deleteRecurringExpense: jest.fn(),
	}
	return shallow(<RecurringExpenseTable {...propsToUser} />)
}
