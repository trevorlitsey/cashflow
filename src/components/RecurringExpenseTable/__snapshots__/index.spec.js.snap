// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RecurringExpenseTable should render without crashing 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <RecurringExpenseTable
    deleteRecurringExpense={[MockFunction]}
    recurringExpenses={
      Object {
        "one": Object {
          "amount": 200,
          "frequency": 2,
          "interval": "days",
          "name": "ya!",
          "startDate": 987628949876,
        },
        "two": Object {
          "amount": 200,
          "frequency": 1,
          "interval": "weeks",
          "name": "check this out",
          "startDate": 9852384982345,
        },
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <styled.ul>
        <li>
          <div
            className="left"
          >
            <p
              className="title"
            >
              ya! ($200)
            </p>
            <p
              className="subtitle"
            >
              every 2 days starting on April 18, 2001
            </p>
          </div>
          <div
            className="right"
          >
            <a>
              edit
            </a>
            |
            <Popconfirm
              cancelText="No"
              okText="Yes"
              okType="primary"
              onConfirm={[Function]}
              placement="top"
              prefixCls="ant-popover"
              title="Are you sure you want to delete this expense?"
              transitionName="zoom-big"
              trigger="click"
            >
              <a
                data-test="delete"
              >
                delete
              </a>
            </Popconfirm>
          </div>
        </li>
        <li>
          <div
            className="left"
          >
            <p
              className="title"
            >
              check this out ($200)
            </p>
            <p
              className="subtitle"
            >
              every 1 weeks starting on March 18, 2282
            </p>
          </div>
          <div
            className="right"
          >
            <a>
              edit
            </a>
            |
            <Popconfirm
              cancelText="No"
              okText="Yes"
              okType="primary"
              onConfirm={[Function]}
              placement="top"
              prefixCls="ant-popover"
              title="Are you sure you want to delete this expense?"
              transitionName="zoom-big"
              trigger="click"
            >
              <a
                data-test="delete"
              >
                delete
              </a>
            </Popconfirm>
          </div>
        </li>
      </styled.ul>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <li>
            <div
              className="left"
            >
              <p
                className="title"
              >
                ya! ($200)
              </p>
              <p
                className="subtitle"
              >
                every 2 days starting on April 18, 2001
              </p>
            </div>
            <div
              className="right"
            >
              <a>
                edit
              </a>
              |
              <Popconfirm
                cancelText="No"
                okText="Yes"
                okType="primary"
                onConfirm={[Function]}
                placement="top"
                prefixCls="ant-popover"
                title="Are you sure you want to delete this expense?"
                transitionName="zoom-big"
                trigger="click"
              >
                <a
                  data-test="delete"
                >
                  delete
                </a>
              </Popconfirm>
            </div>
          </li>,
          <li>
            <div
              className="left"
            >
              <p
                className="title"
              >
                check this out ($200)
              </p>
              <p
                className="subtitle"
              >
                every 1 weeks starting on March 18, 2282
              </p>
            </div>
            <div
              className="right"
            >
              <a>
                edit
              </a>
              |
              <Popconfirm
                cancelText="No"
                okText="Yes"
                okType="primary"
                onConfirm={[Function]}
                placement="top"
                prefixCls="ant-popover"
                title="Are you sure you want to delete this expense?"
                transitionName="zoom-big"
                trigger="click"
              >
                <a
                  data-test="delete"
                >
                  delete
                </a>
              </Popconfirm>
            </div>
          </li>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "one",
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="left"
              >
                <p
                  className="title"
                >
                  ya! ($200)
                </p>
                <p
                  className="subtitle"
                >
                  every 2 days starting on April 18, 2001
                </p>
              </div>,
              <div
                className="right"
              >
                <a>
                  edit
                </a>
                |
                <Popconfirm
                  cancelText="No"
                  okText="Yes"
                  okType="primary"
                  onConfirm={[Function]}
                  placement="top"
                  prefixCls="ant-popover"
                  title="Are you sure you want to delete this expense?"
                  transitionName="zoom-big"
                  trigger="click"
                >
                  <a
                    data-test="delete"
                  >
                    delete
                  </a>
                </Popconfirm>
              </div>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <p
                    className="title"
                  >
                    ya! ($200)
                  </p>,
                  <p
                    className="subtitle"
                  >
                    every 2 days starting on April 18, 2001
                  </p>,
                ],
                "className": "left",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "ya! ($200)",
                    "className": "title",
                  },
                  "ref": null,
                  "rendered": "ya! ($200)",
                  "type": "p",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "every 2 days starting on April 18, 2001",
                    "className": "subtitle",
                  },
                  "ref": null,
                  "rendered": "every 2 days starting on April 18, 2001",
                  "type": "p",
                },
              ],
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <a>
                    edit
                  </a>,
                  "|",
                  <Popconfirm
                    cancelText="No"
                    okText="Yes"
                    okType="primary"
                    onConfirm={[Function]}
                    placement="top"
                    prefixCls="ant-popover"
                    title="Are you sure you want to delete this expense?"
                    transitionName="zoom-big"
                    trigger="click"
                  >
                    <a
                      data-test="delete"
                    >
                      delete
                    </a>
                  </Popconfirm>,
                ],
                "className": "right",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "edit",
                  },
                  "ref": null,
                  "rendered": "edit",
                  "type": "a",
                },
                "|",
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "cancelText": "No",
                    "children": <a
                      data-test="delete"
                    >
                      delete
                    </a>,
                    "okText": "Yes",
                    "okType": "primary",
                    "onConfirm": [Function],
                    "placement": "top",
                    "prefixCls": "ant-popover",
                    "title": "Are you sure you want to delete this expense?",
                    "transitionName": "zoom-big",
                    "trigger": "click",
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "delete",
                      "data-test": "delete",
                    },
                    "ref": null,
                    "rendered": "delete",
                    "type": "a",
                  },
                  "type": [Function],
                },
              ],
              "type": "div",
            },
          ],
          "type": "li",
        },
        Object {
          "instance": null,
          "key": "two",
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="left"
              >
                <p
                  className="title"
                >
                  check this out ($200)
                </p>
                <p
                  className="subtitle"
                >
                  every 1 weeks starting on March 18, 2282
                </p>
              </div>,
              <div
                className="right"
              >
                <a>
                  edit
                </a>
                |
                <Popconfirm
                  cancelText="No"
                  okText="Yes"
                  okType="primary"
                  onConfirm={[Function]}
                  placement="top"
                  prefixCls="ant-popover"
                  title="Are you sure you want to delete this expense?"
                  transitionName="zoom-big"
                  trigger="click"
                >
                  <a
                    data-test="delete"
                  >
                    delete
                  </a>
                </Popconfirm>
              </div>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <p
                    className="title"
                  >
                    check this out ($200)
                  </p>,
                  <p
                    className="subtitle"
                  >
                    every 1 weeks starting on March 18, 2282
                  </p>,
                ],
                "className": "left",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "check this out ($200)",
                    "className": "title",
                  },
                  "ref": null,
                  "rendered": "check this out ($200)",
                  "type": "p",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "every 1 weeks starting on March 18, 2282",
                    "className": "subtitle",
                  },
                  "ref": null,
                  "rendered": "every 1 weeks starting on March 18, 2282",
                  "type": "p",
                },
              ],
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <a>
                    edit
                  </a>,
                  "|",
                  <Popconfirm
                    cancelText="No"
                    okText="Yes"
                    okType="primary"
                    onConfirm={[Function]}
                    placement="top"
                    prefixCls="ant-popover"
                    title="Are you sure you want to delete this expense?"
                    transitionName="zoom-big"
                    trigger="click"
                  >
                    <a
                      data-test="delete"
                    >
                      delete
                    </a>
                  </Popconfirm>,
                ],
                "className": "right",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "edit",
                  },
                  "ref": null,
                  "rendered": "edit",
                  "type": "a",
                },
                "|",
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "cancelText": "No",
                    "children": <a
                      data-test="delete"
                    >
                      delete
                    </a>,
                    "okText": "Yes",
                    "okType": "primary",
                    "onConfirm": [Function],
                    "placement": "top",
                    "prefixCls": "ant-popover",
                    "title": "Are you sure you want to delete this expense?",
                    "transitionName": "zoom-big",
                    "trigger": "click",
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "delete",
                      "data-test": "delete",
                    },
                    "ref": null,
                    "rendered": "delete",
                    "type": "a",
                  },
                  "type": [Function],
                },
              ],
              "type": "div",
            },
          ],
          "type": "li",
        },
      ],
      "type": [Function],
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <styled.ul>
          <li>
            <div
              className="left"
            >
              <p
                className="title"
              >
                ya! ($200)
              </p>
              <p
                className="subtitle"
              >
                every 2 days starting on April 18, 2001
              </p>
            </div>
            <div
              className="right"
            >
              <a>
                edit
              </a>
              |
              <Popconfirm
                cancelText="No"
                okText="Yes"
                okType="primary"
                onConfirm={[Function]}
                placement="top"
                prefixCls="ant-popover"
                title="Are you sure you want to delete this expense?"
                transitionName="zoom-big"
                trigger="click"
              >
                <a
                  data-test="delete"
                >
                  delete
                </a>
              </Popconfirm>
            </div>
          </li>
          <li>
            <div
              className="left"
            >
              <p
                className="title"
              >
                check this out ($200)
              </p>
              <p
                className="subtitle"
              >
                every 1 weeks starting on March 18, 2282
              </p>
            </div>
            <div
              className="right"
            >
              <a>
                edit
              </a>
              |
              <Popconfirm
                cancelText="No"
                okText="Yes"
                okType="primary"
                onConfirm={[Function]}
                placement="top"
                prefixCls="ant-popover"
                title="Are you sure you want to delete this expense?"
                transitionName="zoom-big"
                trigger="click"
              >
                <a
                  data-test="delete"
                >
                  delete
                </a>
              </Popconfirm>
            </div>
          </li>
        </styled.ul>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <li>
              <div
                className="left"
              >
                <p
                  className="title"
                >
                  ya! ($200)
                </p>
                <p
                  className="subtitle"
                >
                  every 2 days starting on April 18, 2001
                </p>
              </div>
              <div
                className="right"
              >
                <a>
                  edit
                </a>
                |
                <Popconfirm
                  cancelText="No"
                  okText="Yes"
                  okType="primary"
                  onConfirm={[Function]}
                  placement="top"
                  prefixCls="ant-popover"
                  title="Are you sure you want to delete this expense?"
                  transitionName="zoom-big"
                  trigger="click"
                >
                  <a
                    data-test="delete"
                  >
                    delete
                  </a>
                </Popconfirm>
              </div>
            </li>,
            <li>
              <div
                className="left"
              >
                <p
                  className="title"
                >
                  check this out ($200)
                </p>
                <p
                  className="subtitle"
                >
                  every 1 weeks starting on March 18, 2282
                </p>
              </div>
              <div
                className="right"
              >
                <a>
                  edit
                </a>
                |
                <Popconfirm
                  cancelText="No"
                  okText="Yes"
                  okType="primary"
                  onConfirm={[Function]}
                  placement="top"
                  prefixCls="ant-popover"
                  title="Are you sure you want to delete this expense?"
                  transitionName="zoom-big"
                  trigger="click"
                >
                  <a
                    data-test="delete"
                  >
                    delete
                  </a>
                </Popconfirm>
              </div>
            </li>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": "one",
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <div
                  className="left"
                >
                  <p
                    className="title"
                  >
                    ya! ($200)
                  </p>
                  <p
                    className="subtitle"
                  >
                    every 2 days starting on April 18, 2001
                  </p>
                </div>,
                <div
                  className="right"
                >
                  <a>
                    edit
                  </a>
                  |
                  <Popconfirm
                    cancelText="No"
                    okText="Yes"
                    okType="primary"
                    onConfirm={[Function]}
                    placement="top"
                    prefixCls="ant-popover"
                    title="Are you sure you want to delete this expense?"
                    transitionName="zoom-big"
                    trigger="click"
                  >
                    <a
                      data-test="delete"
                    >
                      delete
                    </a>
                  </Popconfirm>
                </div>,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <p
                      className="title"
                    >
                      ya! ($200)
                    </p>,
                    <p
                      className="subtitle"
                    >
                      every 2 days starting on April 18, 2001
                    </p>,
                  ],
                  "className": "left",
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "ya! ($200)",
                      "className": "title",
                    },
                    "ref": null,
                    "rendered": "ya! ($200)",
                    "type": "p",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "every 2 days starting on April 18, 2001",
                      "className": "subtitle",
                    },
                    "ref": null,
                    "rendered": "every 2 days starting on April 18, 2001",
                    "type": "p",
                  },
                ],
                "type": "div",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <a>
                      edit
                    </a>,
                    "|",
                    <Popconfirm
                      cancelText="No"
                      okText="Yes"
                      okType="primary"
                      onConfirm={[Function]}
                      placement="top"
                      prefixCls="ant-popover"
                      title="Are you sure you want to delete this expense?"
                      transitionName="zoom-big"
                      trigger="click"
                    >
                      <a
                        data-test="delete"
                      >
                        delete
                      </a>
                    </Popconfirm>,
                  ],
                  "className": "right",
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "edit",
                    },
                    "ref": null,
                    "rendered": "edit",
                    "type": "a",
                  },
                  "|",
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "class",
                    "props": Object {
                      "cancelText": "No",
                      "children": <a
                        data-test="delete"
                      >
                        delete
                      </a>,
                      "okText": "Yes",
                      "okType": "primary",
                      "onConfirm": [Function],
                      "placement": "top",
                      "prefixCls": "ant-popover",
                      "title": "Are you sure you want to delete this expense?",
                      "transitionName": "zoom-big",
                      "trigger": "click",
                    },
                    "ref": null,
                    "rendered": Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "children": "delete",
                        "data-test": "delete",
                      },
                      "ref": null,
                      "rendered": "delete",
                      "type": "a",
                    },
                    "type": [Function],
                  },
                ],
                "type": "div",
              },
            ],
            "type": "li",
          },
          Object {
            "instance": null,
            "key": "two",
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <div
                  className="left"
                >
                  <p
                    className="title"
                  >
                    check this out ($200)
                  </p>
                  <p
                    className="subtitle"
                  >
                    every 1 weeks starting on March 18, 2282
                  </p>
                </div>,
                <div
                  className="right"
                >
                  <a>
                    edit
                  </a>
                  |
                  <Popconfirm
                    cancelText="No"
                    okText="Yes"
                    okType="primary"
                    onConfirm={[Function]}
                    placement="top"
                    prefixCls="ant-popover"
                    title="Are you sure you want to delete this expense?"
                    transitionName="zoom-big"
                    trigger="click"
                  >
                    <a
                      data-test="delete"
                    >
                      delete
                    </a>
                  </Popconfirm>
                </div>,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <p
                      className="title"
                    >
                      check this out ($200)
                    </p>,
                    <p
                      className="subtitle"
                    >
                      every 1 weeks starting on March 18, 2282
                    </p>,
                  ],
                  "className": "left",
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "check this out ($200)",
                      "className": "title",
                    },
                    "ref": null,
                    "rendered": "check this out ($200)",
                    "type": "p",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "every 1 weeks starting on March 18, 2282",
                      "className": "subtitle",
                    },
                    "ref": null,
                    "rendered": "every 1 weeks starting on March 18, 2282",
                    "type": "p",
                  },
                ],
                "type": "div",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <a>
                      edit
                    </a>,
                    "|",
                    <Popconfirm
                      cancelText="No"
                      okText="Yes"
                      okType="primary"
                      onConfirm={[Function]}
                      placement="top"
                      prefixCls="ant-popover"
                      title="Are you sure you want to delete this expense?"
                      transitionName="zoom-big"
                      trigger="click"
                    >
                      <a
                        data-test="delete"
                      >
                        delete
                      </a>
                    </Popconfirm>,
                  ],
                  "className": "right",
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "edit",
                    },
                    "ref": null,
                    "rendered": "edit",
                    "type": "a",
                  },
                  "|",
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "class",
                    "props": Object {
                      "cancelText": "No",
                      "children": <a
                        data-test="delete"
                      >
                        delete
                      </a>,
                      "okText": "Yes",
                      "okType": "primary",
                      "onConfirm": [Function],
                      "placement": "top",
                      "prefixCls": "ant-popover",
                      "title": "Are you sure you want to delete this expense?",
                      "transitionName": "zoom-big",
                      "trigger": "click",
                    },
                    "ref": null,
                    "rendered": Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "children": "delete",
                        "data-test": "delete",
                      },
                      "ref": null,
                      "rendered": "delete",
                      "type": "a",
                    },
                    "type": [Function],
                  },
                ],
                "type": "div",
              },
            ],
            "type": "li",
          },
        ],
        "type": [Function],
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
